### PR TITLE
[TxPool] Reject future transactions during high memory usage

### PR DESCRIPTION
**Pull request by dbrajovic**
# Description

This PR adds an additional constraint on transaction acquisition during periods of high slots usage.

If a transaction sent to the pool is not expected in terms of nonce (meaning it's a future transactions or spam) then it is rejected. 

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

# Additional comments

Fixes EDGE-732

### CODE REVIEW

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯

**Review by 0xAleksaOpacic — APPROVED** [review]
LGTM, tested with SportX script on local everything works

File: txpool/txpool.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ var (
 	ErrAlreadyKnown            = errors.New("already known")
 	ErrOversizedData           = errors.New("oversized data")
 	ErrMaxEnqueuedLimitReached = errors.New("maximum number of enqueued transactions reached")
+	ErrRejectFutureTx          = errors.New("rejected future tx due to low memory")
```
Comment (Kourin1996) [inline txpool/txpool.go]: Why memory instead of the term, slots? I don't think memory usage is related to this error

**Review by Kourin1996 — COMMENTED** [review]
Just a small comment

File: txpool/txpool.go (hunk 1 above)
Comment (dbrajovic) [reply txpool/txpool.go]: Well, slots is the pool's concept of memory. 

I'll rename it to slots.
[Renamed](https://github.com/0xPolygon/polygon-edge/pull/690/commits/7f66b77089d1b2d378435a40a9100cd05888110a)

**Review by Kourin1996 — APPROVED** [review]

**Review by JurbinM — APPROVED** [review]

**Pull request merged**